packages/@aws-cdk/aws-stepfunctions/test/states-language.test.ts

```diff
@@ -614,6 +614,26 @@ describe('States Language', () => {
     expect(() => new stepfunctions.Parallel(stack, 'Parallel')
       .branch(state1.next(state2))
       .branch(state2)).toThrow();
+  }),
+
+  test('Can retrieve possible states from initial state', () => {
```

Comment:
Need more test cases here - 
1. Assert that this behaves correctly when there is a Choice and Parallel states included.
2. Assert that states that are not reachable are not returned.

Wrap these test cases inside of a describe clause - 

```ts
describe('findReachableStates', () => {
  test('Can retrieve possible states from initial state', () => {
    ...
  });
});
```

Reply:
The construct sees the Parallel subsequent states as separate State Machines, so none of the private methods used to compute the subsequent states actually care about them. It looks to me like a design decision that is out of the scope of this PR. 
Therefore, the states inside a Parallel states are not retrieved.

Reply:
I'm not so sure that this is a conscious design choice. 

It seems to me that `outgoingTransitions()` should take an additional option called `includeBranches` (which defaults to `false`) and considers all the branches (`this.state`) of the current state as well.

Otherwise, this utility function is incomplete.

cc @rix0rrr.

Reply:
Parallel branches are separate state machines, as far as I'm aware, so yes this was done consciously. Let me read the rest of the thread.

Reply:
> I'm not so sure that this is a conscious design choice.
> 
> It seems to me that `outgoingTransitions()` should take an additional option called `includeBranches` (which defaults to `false`) and considers all the branches (`this.state`) of the current state as well.
> 
> Otherwise, this utility function is incomplete.
> 
> cc @rix0rrr.

I like this approach better. Please let me know if I should add it in this PR.

Reply:
Given that we don't need this yet, let's leave the implementation as is, and add explicit documentation to the tsdoc that these states are not returned.

Reply:
Ok!